### DISCUSSION TITLE
Documentation on renderToTimemap options

### DISCUSSION THREAD

**Discussion by rettinghaus** [rism-digital/verovio#3480] [Q&A]
AFAICS the timemap options are not documented in the reference book, correct?

**Comment by craigsapp** [comment]
There is documentation on page 77 of the PDF:

https://book.verovio.org/verovio-reference-book.pdf#page=77

It took me a while to figure out how to get the timemap from the command-line tool with which I was checking out the missed tempos a few weeks ago because I was searching the options (and not checking in the reference manual).  So it might be nice to add `timemap` to the `--output-to` documentation in the `verovio -h` list of options which is currently:

```
 -t, --output-to <s>            Select output format to: "mei", "mei-pb", "mei-basic", "svg", or "midi" (default: "svg")
```

Also `humdrum` is another possible output (but is still experimental and may not work for complicated scores, so best to leave out for now).

**Comment by craigsapp** [comment]
`tools/main.cpp` error message has a more complete list of output formats which might be nice to duplicate in the `verovio -h` output:

```cpp
        std::cerr << "Output format (" << outformat
                  << ") can only be 'mei', 'mei-basic', 'mei-pb', 'svg', 'midi', 'timemap', 'expansionmap', 'humdrum' "
                     "or 'pae'."
                  << std::endl;
```

**Comment by rettinghaus** [comment]
I was asking about the time map options as seen here: https://github.com/rism-digital/verovio/issues/2756

**Comment by craigsapp** [comment]
Yes, I was just checking on that in the code (I have not used them yet, and I don't think they are listed in the documentation either):

`src/toolkit.cpp`:

```cpp
std::string Toolkit::RenderToTimemap(const std::string &jsonOptions) 
{       
    bool includeMeasures = false;
    bool includeRests = false;
    
    jsonxx::Object json;
        
    // Read JSON options if not empty
    if (!jsonOptions.empty()) {
        if (!json.parse(jsonOptions)) {
            LogWarning("Cannot parse JSON std::string. Using default options.");
        }
        else {
            if (json.has<jsonxx::Boolean>("includeMeasures"))
                includeMeasures = json.get<jsonxx::Boolean>("includeMeasures");
            if (json.has<jsonxx::Boolean>("includeRests")) includeRests = json.get<jsonxx::Boolean>("includeRests");
        }
    }

    this->ResetLogBuffer();

    std::string output;
    m_doc.ExportTimemap(output, includeRests, includeMeasures);
    return output;
}
```

So currently `includeMeasures` and `includeRests`.
